pkg/netflow/flowaggregator/aggregator_test.go

```diff
@@ -0,0 +1,131 @@
+package flowaggregator
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/mocksender"
+	coreconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/netflow/common"
+	"github.com/DataDog/datadog-agent/pkg/netflow/config"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"net"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestAggregator(t *testing.T) {
+	stoppedMu := sync.RWMutex{} // Mutex needed to avoid race condition in test
+
+	coreconfig.Datadog.Set("hostname", "my-hostname")
+	sender := mocksender.NewMockSender("")
+	sender.On("Count", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return()
+	sender.On("EventPlatformEvent", mock.Anything, mock.Anything).Return()
+	sender.On("Commit").Return()
+	conf := config.NetflowConfig{
+		StopTimeout:             10,
+		AggregatorBufferSize:    20,
+		AggregatorFlushInterval: 1,
+		LogPayloads:             true,
+		Listeners: []config.ListenerConfig{
+			{
+				FlowType: common.TypeNetFlow9,
+				BindHost: "127.0.0.1",
+				Port:     uint16(1234),
+				Workers:  10,
+			},
+		},
+	}
+	flow := &common.Flow{
+		Namespace:      "my-ns",
+		FlowType:       common.TypeNetFlow9,
+		ExporterAddr:   net.IP([]byte{127, 0, 0, 1}).String(),
+		StartTimestamp: 1234568,
+		EndTimestamp:   1234569,
+		Bytes:          20,
+		Packets:        4,
+		SrcAddr:        net.IP([]byte{10, 10, 10, 10}).String(),
+		DstAddr:        net.IP([]byte{10, 10, 10, 20}).String(),
+		IPProtocol:     uint32(6),
+		SrcPort:        uint32(2000),
+		DstPort:        uint32(80),
+	}
+
+	aggregator := NewFlowAggregator(sender, &conf)
+	aggregator.flushInterval = 1 * time.Second
+	inChan := aggregator.GetFlowInChan()
+
+	expectStartExisted := false
+	go func() {
+		aggregator.Start()
+		stoppedMu.Lock()
+		expectStartExisted = true
+		stoppedMu.Unlock()
+	}()
+	inChan <- flow
+
+	time.Sleep(3 * time.Second)
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
Same here, I recommend using a smaller polling interval with an eventual timeout. Sleeping tests for 3 seconds is not ideal

Reply:
Fixed here https://github.com/DataDog/datadog-agent/pull/10976/commits/7da5b73e186f6d4ec1f539a22310815bbd777914